### PR TITLE
Added device name to navbar.  Changed network-interfaces page to network

### PR DESCRIPTION
**Pull request by Davo36**
Closes #534

### CODE REVIEW

File: management-interface.go [hunk 1]
```diff
@@ -49,17 +52,36 @@ var tmpl *template.Template
...
+	deviceName = getDeviceName()
+
 	tmpl = template.New("")
 
 	for _, name := range templateBox.List() {
 		t := tmpl.New(name)
-		template.Must(t.Parse(templateBox.String(name)))
+		if name == "navbar.html" {
+			stringToParse := templateBox.String(name)
+			stringToParse = strings.Replace(stringToParse, "Cacophonator Management", deviceName, 1)
```
Comment (mjs) [inline management-interface.go]: It's a little odd to do this considering that we're already using templates!

It should be possible to pass the hostname into each ExecuteTemplate call with a little bit of extra work. It's OK to leave this for now though. If you leave it, please a comment about doing this better.

Comment (Davo36) [reply management-interface.go]: Hmmm yes.  Ironic I know.

But if I don't do it there, then each call to ExecuteTemplate from each of the handlers would have to return that deviceName variable wouldn't it?

And some of the handlers return nil, and some of them return structures with data.  So I'd need to make them all return that device name inside a structure and use something like {{.DeviceName}} in the navbar template.

I couldn't really see how to just write that value into a header template once at startup.  I could do it in the init function, but ExecuteTemplate takes a writer as a parameter and I didn't know what to do with that.

Comment (Davo36) [reply management-interface.go]: Menno, I did find a better way to do this.  I map a function to the template tree.  It is called from the navbar.html file when needed.  Hope that is better?

File: management-interface.go [hunk 2]
```diff
@@ -49,7 +52,11 @@ var tmpl *template.Template
 // This does some initialisation.  It parses our html templates up front and
 // finds the location where this executable was started.
 func init() {
+
+	setDeviceName()
+
 	tmpl = template.New("")
+	tmpl.Funcs(template.FuncMap{"DeviceName": getDeviceName})
```
Comment (mjs) [inline management-interface.go]: Using a template function is nice workaround.

You can avoid the need for the `deviceName` module level variable and the current getDeviceName function by using a closure. Have setDeviceName just return the device name (and perhaps call it getDeviceName!) and then do this here:

```go
deviceName := getDeviceName()
tmpl.Funcs(template.FuncMap{"DeviceName": func() { return deviceName }})
```

File: management-interface.go (hunk 1 above)
Comment (mjs) [reply management-interface.go]: Yep that's better. I was thinking you could have a function which generates the global variables to pass to each ExecuteTemplate call which then somehow get combined with any specific values for the template, but using a function is almost certainly simpler.

**Comment by Davo36** [comment]
As usual, your ideas are sound.  Is this as you want it now?

File: management-interface.go [hunk 3]
```diff
@@ -67,28 +64,17 @@ func init() {
...
-		deviceName = "Unknown Device."
-	} else {
-		deviceName = name
-		// Check for the case when name could be something like: 'host.corp.com'
-		// If it is, just use the part before the first dot.
-		pos := strings.Index(name, ".")
-		if pos != -1 {
-			deviceName = name[:pos]
-		}
+		return "Unknown Device."
```
Comment (mjs) [inline management-interface.go]: This should probably just be "Unknown"

**Pull request merged**